src/DistributedArrays.jl

```diff
@@ -139,6 +132,104 @@ macro DArray(ex::Expr)
                 tuple($(map(r->:(length($r)), ranges)...))) )
 end
 
+# new DArray similar to an existing one
+DArray(init, d::DArray) = DArray(init, size(d), procs(d), [size(d.pids)...])
+
+function release_darray(d::DArray)
+    if (myid() == d.identity[1]) && d.release
+        @schedule close(d.identity, d.pids)
+        d.release = false
+    end
+end
+
+function construct_darray(identity, init, dims, procs, idxs, cuts)
+    r=RemoteRef()
+    @sync begin
+        for i = 1:length(procs)
+            @async begin
+                local typA
+                if isa(init, Function)
+                    typA=remotecall_fetch(construct_localparts, procs[i], init, identity, dims, procs, idxs, cuts)
+                else
+                    typA=remotecall_fetch(construct_localparts, procs[i], init[i], identity, dims, procs, idxs, cuts)
+                end
+                !isready(r) && put!(r, typA)
+            end
+        end
+    end
+
+    typA = take!(r)
+    if myid() in procs
+        return registry[(identity, :DARRAY)]
+    else
+        return DArray{eltype(typA),length(dims),typA}(identity, dims, procs, idxs, cuts)
+    end
+end
+
+function construct_localparts(init, identity, dims, procs, idxs, cuts)
+    A = isa(init, Function) ? init(idxs[localpartindex(procs)]) : fetch(init)
+    global registry
+    registry[(identity, :LOCALPART)] = A
+    typA = typeof(A)
+    d = DArray{eltype(typA),length(dims),typA}(identity, dims, procs, idxs, cuts)
+    registry[(identity, :DARRAY)] = d
+    typA
+end
+
+let DID::Int = 1
```

Comment:
Wouldn't the identity have to consistent across all processes?  How does it work when you create a darray on other processes other than process 1?  Ex.  An array shared across all processes, one created on process 1, process 2, etc...